src/lib.rs

```diff
@@ -428,7 +428,11 @@ pub fn get_tx_timestamp(sock: &TsnSocket) -> Result<time::Timespec, Error> {
             let ts = match ts {
                 ts if ts[2].tv_sec != 0 || ts[2].tv_nsec != 0 => ts[2],
                 ts if ts[1].tv_sec != 0 || ts[1].tv_nsec != 0 => ts[1],
-                ts if ts[0].tv_sec != 0 || ts[0].tv_nsec != 0 => ts[0],
+                ts if ts[0].tv_sec != 0 || ts[0].tv_nsec != 0 => {
+                    // Log warning
+                    eprintln!("SW TX timestamp(from driver) used");
```

Comment:
Doesn't it print error every time the SW TX timestamp is loaded?
If can't use HW TX timestamp, too many printed error message

Reply:
Yes, it prints every time when SW TS is used.

But while HW TS is available, low chance of SW timestamp being used. This is for that case.
If HW TS is not available at all, It is not too bad to print it every time. And user could get rid of it by redirect stderr into null (`command 2>/dev/null`)